test/e2e/cloud/instance_start_test.go

```diff
@@ -0,0 +1,337 @@
+// Copyright (c) 2024, Unikraft GmbH and The KraftKit Authors.
+// Licensed under the BSD-3-Clause License (the "License").
+// You may not use this file except in compliance with the License.
+
+package cloud_test
+
+import (
+	"crypto/rand"
+	"fmt"
+	"math/big"
+	"os"
+
+	. "github.com/onsi/ginkgo/v2" //nolint:stylecheck
+	. "github.com/onsi/gomega"    //nolint:stylecheck
+
+	fcmd "kraftkit.sh/test/e2e/framework/cmd"
+	fcfg "kraftkit.sh/test/e2e/framework/config"
+)
+
+var _ = Describe("kraft cloud vm start", func() {
+	var cmd *fcmd.Cmd
+
+	var stdout *fcmd.IOStream
+	var stderr *fcmd.IOStream
+
+	var cfg *fcfg.Config
+	var instanceNameFull string
+
+	const (
+		imageName      = "nginx:latest"
+		instanceName   = "instance-start-test"
+		instanceMemory = "64"
+	)
+
+	BeforeEach(func() {
+		token := os.Getenv("UNIKRAFTCLOUD_TOKEN")
+
+		if token == "" {
+			token = os.Getenv("KRAFTCLOUD_TOKEN")
+		}
+
+		if token == "" {
+			token = os.Getenv("KC_TOKEN")
+		}
+
+		if token == "" {
+			token = os.Getenv("UKC_TOKEN")
+		}
+
+		if token == "" {
+			Skip("UNIKRAFTCLOUD_TOKEN is not set")
+		}
+
+		metro := os.Getenv("UNIKRAFTCLOUD_METRO")
+
+		if metro == "" {
+			metro = os.Getenv("KRAFTCLOUD_METRO")
+		}
+
+		if metro == "" {
+			metro = os.Getenv("KC_METRO")
+		}
+
+		if metro == "" {
+			metro = os.Getenv("UKC_METRO")
+		}
+
+		if metro == "" {
+			Skip("UNIKRAFTCLOUD_METRO is not set")
+		}
+
+		stdout = fcmd.NewIOStream()
+		stderr = fcmd.NewIOStream()
+
+		cfg = fcfg.NewTempConfig()
+
+		createCmd := fcmd.NewKraft(stdout, stderr, cfg.Path())
+		createCmd.Env = os.Environ()
+		createCmd.Args = append(createCmd.Args, "cloud", "instance", "create", "-p", "11111:8080/tls", "--log-level", "info", "--log-type", "json", "-o", "json")
+
+		id, err := rand.Int(rand.Reader, big.NewInt(100000000000))
+		if err != nil {
+			panic(err)
+		}
+		instanceNameFull = fmt.Sprintf("%s-%d", instanceName, id)
+
+		createCmd.Args = append(createCmd.Args,
+			"--memory", instanceMemory,
+			"--name", instanceNameFull,
+			imageName,
+		)
+
+		err = createCmd.Run()
+		if err != nil {
+			fmt.Print(createCmd.DumpError(stdout, stderr, err))
+		}
+
+		Expect(err).ToNot(HaveOccurred())
+		Expect(stderr.String()).To(BeEmpty())
+		Expect(stdout.String()).To(MatchRegexp(`stopped`))
+
+		stdout = fcmd.NewIOStream()
+		stderr = fcmd.NewIOStream()
+
+		cmd = fcmd.NewKraft(stdout, stderr, cfg.Path())
+		cmd.Env = os.Environ()
+		cmd.Args = append(cmd.Args, "cloud", "vm", "start", "--log-level", "info", "--log-type", "json")
+	})
+
+	AfterEach(func() {
+		rmCmd := fcmd.NewKraft(stdout, stderr, cfg.Path())
+		rmCmd.Env = os.Environ()
+		rmCmd.Args = append(rmCmd.Args, "cloud", "vm", "rm", "--log-level", "info", "--log-type", "json", instanceNameFull)
+
+		err := rmCmd.Run()
+		if err != nil {
+			fmt.Print(rmCmd.DumpError(stdout, stderr, err))
+		}
+
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	When("invoked with the instance name and the wait flag", func() {
```

Comment:
same as with `restart` - i would also want to test that the flag causes a timeout